Userland/Libraries/LibGfx/ImageFormats/JPEGXLLoader.cpp

```diff
@@ -1391,6 +1391,48 @@ class SelfCorrectingData {
 ///
 
 /// H.2 - Image decoding
+
+static ErrorOr<void> add_default_squeeze_params(TransformInfo& tr, Span<ChannelInfo> channels, u32 nb_meta_channels)
+{
+    // H.6.2.1  Parameters - "The default parameters (the case when sp.size() == 0) are specified by the following code:"
+
+    auto first = nb_meta_channels;
+    auto count = channels.size() - first;
+    auto w = channels[first].width;
+    auto h = channels[first].height;
+    SqueezeParams param;
+    if (count > 2 && channels[first + 1].width == w && channels[first + 1].height == h) {
+        param.begin_c = first + 1;
+        param.num_c = 2;
+        param.in_place = false;
+        param.horizontal = true;
+        tr.sp.append(param);
+        param.horizontal = false;
+        tr.sp.append(param);
+    }
+    param.begin_c = first;
+    param.num_c = count;
+    param.in_place = true;
+    if (h >= w && h > 8) {
+        param.horizontal = false;
+        tr.sp.append(param);
+        h = (h + 1) / 2;
+    }
+    while (w > 8 || h > 8) {
+        if (w > 8) {
+            param.horizontal = true;
+            tr.sp.append(param);
+            w = (w + 1) / 2;
```

Comment:
`w = ceil_div(w, 2)`

(also below for `h`)

Reply:
This comes from the spec.